vault/cors.go

```diff
@@ -0,0 +1,137 @@
+package vault
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var errCORSNotConfigured = errors.New("CORS is not configured")
+
+type CORSConfig struct {
+	enabled          bool
+	allowedOrigins   *[]string
+	allowedHeaders   *map[string]string
```

Comment:
What is the point of these other struct members considering that they cannot be changed?